File: src/server/lib/geo/rest/metadata.py
from bottle import *
from geo.model import *
import geo.rest.vo as vo

@route('/metadata/distinct_values')
def distinct_values():
	name = request.GET.get('name')
	layer_name = request.GET.get('layer')
	limit = request.GET.get('limit')
	m = Metadata(name)
	values = m.distinct_values(layer_name, limit)
	return vo.collection(values, len(values))

File: src/js-api/geo-core/src/core/metadata.js
Ext.ns("Geo.core");
Geo.core.Metadata = Ext.extend(Ext.util.Observable, {
	constructor: function (config) {
		this.addEvents(
			'distinct_values'
		);
		Geo.core.Metadata.superclass.constructor.call(config);
	}
	,
	getDistinctValues:function(name, layer, limit){

		var p = {
			name : name,
			layer : layer,
			limit : limit
		};

		Ext.Ajax.request({
			url: Geo.UriTemplate.getUri('metadataDistinctValues', '?' + Ext.urlEncode(p)),
			method: 'Get',
			headers: { 'Content-Type': 'text/json' },
			scope: this,
			success: function (response, options) {
				var result = Ext.util.JSON.decode(response.responseText);
				this.fireEvent('distinct_values', result);
			}
		});
	}
});